guides/advanced_guides/authentication.md

```diff
@@ -1,34 +1,34 @@
-# Keys
+# Authentication
 
-In MeiliSearch, there are three types of keys:
+MeiliSearch uses a key-based authentication. There are three types of keys:
 
-- The **Master** key has access to all routes.
-- The **Private** key has access to all routes except the `/keys` routes.
-- The **Public** key has only access to the following routes:
+- The **Master** key grants access to all routes.
+- The **Private** key grants access to all routes except the `/keys` routes.
+- The **Public** key grants only access to the following routes:
   - `GET /indexes/:index_uid/search`
   - `GET /indexes/:index_uid/documents`
   - `GET /indexes/:index_uid/documents/:doc_id`
 
-When a master key is given to MeiliSearch, the engine will automatically generate the private and the public key. **You cannot create additional keys**.
+When a master key is provided to MeiliSearch, both the private and the public keys are automatically generated. **You cannot create any additional keys**.
 
 ### Master Key
 
-When launching an instance of MeiliSearch, you have the possibility to give a master key. By providing a master key, all routes in MeiliSearch will be protected and will require a key to access.
+When launching an instance, you have the option of giving a master key. By doing so, all routes will be protected and will require a key to be accessed.
 
-You can specify this key by passing the `MEILI_MASTER_KEY` environment variable, or with the command line argument `--master-key`.
+You can specify it by passing the `MEILI_MASTER_KEY` environment variable, or using the command line argument `--master-key`.
 
-With this master key, you can retrieve the private, and the public keys using the [keys route](/references/keys.md).
+Thanks to the master key, you can retrieve both the private and the public keys using the [keys route](/references/keys.md).
 
-On each API call, the key must be added to [the header](/references/#authentication).
+On each API call, the master key must be added to [the header](/references/#authentication).
```

Comment:
That's a different sentence. Since you have 3 keys, you can put either one of the keys in the header. 
If you put the public key, you will be able to use the /search route
If you put the private key you will be able to use all the routes expect the key route
and if you use the master key you will be able to use all routes.
```suggestion
## API Key

If a master key is set, on each API call, a key must be added to [the header](/references/#authentication).
If no or a wrong API key is provided in the header you will have no access to any route and you will receive the
`HTTP/1.1 403 Forbidden` status code.

```

If you like this suggestion you could put it under #### No master key

Reply:
Yes, sure! Do I add `## API Key` somewhere?

Reply:
Maybe just before ## Reset Keys ?
Oh and by the way, could you change `### Master Key` and `### Reset Keys` to `## Master Key` and  `## Reset Key`

Reply:
Done!
And removed the former line:

```
On each API call, the master key must be added to [the header](/references/#authentication).
```